File: c_leetcode/lengthOfLastWord.c
#include "stdio.h"
#include <assert.h>
#include "string.h"

int lengthOfLastWord(char* s)
{
    int ans = 0;
    int length = strlen(s) - 1;

    while (length >= 0) {
        if (s[length] == ' ' && ans != 0) {
            break;
        }
        if (s[length] != ' ') {
            ans++;
        }
        length--;
    }
    return ans;
}

int main()
{
    assert(lengthOfLastWord("Hello World") == 5);
    assert(lengthOfLastWord("   fly me   to   the moon  ") == 4);
    assert(lengthOfLastWord("luffy is still joyboy") == 6);

    return 0;
}

File: rust_leetcode/src/bin/length_of_last_word.rs
struct Solution;

impl Solution {
    pub fn length_of_last_word(s: String) -> i32 {
        s.trim()
            .to_string()
            .split_ascii_whitespace()
            .last()
            .unwrap()
            .len() as i32
    }
}

fn main() {
    let s = "Hello World".to_string();
    println!("{}", Solution::length_of_last_word(s));
}

#[cfg(test)]
mod tests {
    use crate::Solution;
    #[test]
    fn test() {
        assert_eq!(Solution::length_of_last_word("Hello World".to_string()), 5);
        assert_eq!(
            Solution::length_of_last_word("   fly me   to   the moon  ".to_string()),
            4
        );
        assert_eq!(
            Solution::length_of_last_word("luffy is still joyboy".to_string()),
            6
        );
    }
}
